### ISSUE TITLE
Add WS server for replicating dat archives / DatArchive polyfill

### ISSUE THREAD

**Issue by RangerMauve** [beakerbrowser/hashbase#96]
Beaker is amazing, but sadly, it's the only one of it's kind at the moment, and not everyone is willing or able to switch to it as their main browser.

Hashbase sits in a great position to enable apps that are made for Beaker to work on the regular Web.

A great start to this is serving static content through HTTPS, but sadly this isn't enough for interactive websites that make use of the DatArchive API.

From experimenting with getting DatArchive working on the web, I've found that all that's really needed is a way to replicate a client-side Hyperdrive instance to the rest of the swarm through a websocket.

This is a fairly [small addition](https://github.com/RangerMauve/dat-gateway/blob/master/index.js#L104) and will probably fit in with the existing codebase without too much added effort, but will enable a broader variety of apps from the Dat ecosystem to work.

Does this sound like something you'd be interested in getting a PR for?

**Comment by pfrazee** [comment]
I'm concerned about the bandwidth costs. We've explicitly avoided running an open gateway because we don't want to go broke!

**Comment by RangerMauve** [comment]
Have you spoken to the IPFS folkes before about their gateways?

What if the gateway detected the origin of the request in order to include the bandwidth as part of the billing for the dat? Then you could have some amount of data free, and application authors would have to pay for more bandwidth.

**Comment by pfrazee** [comment]
I have, it definitely costs them a lot of money and effort.

We can look into creating a "gateway service" where app devs setup API keys and etc, but that's going to take work to do. Not sure when we'd have time for it.

**Comment by RangerMauve** [comment]
Yeah, I don't think this is an urgent at all. Just wanted to bring this up to start discussions.

It'd probably be more relevant months down the line once dat-polyfill is more stable and starts actually getting used by developers.
